android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -847,14 +1094,85 @@ private WritableArray getVideoTrackInfo() {
                 videoTrack.putString("codecs", format.codecs != null ? format.codecs : "");
-                videoTrack.putString("trackId",
-                        format.id == null ? String.valueOf(trackIndex) : format.id);
-                videoTracks.pushMap(videoTrack);
+                videoTrack.putString("trackId", format.id == null ? String.valueOf(trackIndex) : format.id);
+                if (isFormatSupported(format)) {
+                    videoTracks.pushMap(videoTrack);
+                }
             }
         }
+
         return videoTracks;
     }
 
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timeline) {
+        return this.getVideoTrackInfoFromManifest(timeline, 0);
+    }
+
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timelineRef, int retryCount) {
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        final DataSource dataSource = this.mediaDataSourceFactory.createDataSource();
+        final Uri sourceUri = this.srcUri;
+        final long startTime = this.contentStartTime * 1000 - 100; // s -> ms with 100ms offset
+
+        Future<WritableArray> result = es.submit(new Callable<WritableArray>() {
+            DataSource ds = dataSource;
+            Uri uri = sourceUri;
+            Timeline timeline = timelineRef;
+            long startTimeUs = startTime * 1000; // ms -> us
+
+            public WritableArray call() throws Exception {
+                WritableArray videoTracks = Arguments.createArray();
+                try  {
+                    DashManifest manifest = DashUtil.loadManifest(this.ds, this.uri);
+                    int periodCount = manifest.getPeriodCount();
+                    for (int i = 0; i < periodCount; i++) {
+                        Period period = manifest.getPeriod(i);
+                        for (int adaptationIndex = 0; adaptationIndex < period.adaptationSets.size(); adaptationIndex++) {
+                            AdaptationSet adaptation = period.adaptationSets.get(adaptationIndex);
+                            if (adaptation.type != C.TRACK_TYPE_VIDEO) {
+                                continue;
+                            }
+                            boolean hasFoundContentPeriod = false;
+                            for (int representationIndex = 0; representationIndex < adaptation.representations.size(); representationIndex++) {
+                                Representation representation = adaptation.representations.get(representationIndex);
+                                Format format = representation.format;
+                                if (representation.presentationTimeOffsetUs <= startTimeUs) {
+                                    break;
+                                }
+                                hasFoundContentPeriod = true;
+                                WritableMap videoTrack = Arguments.createMap();
```

Comment:
can you avoid creating WritableMap is !isFormatSupported please ?
Additioannal question, can this can be factorized with getVideoTrackInfo ?

Reply:
What do you mean by avoiding WritableMap here? We are returning the video tracks back to JS thats why I think we used it here.
Also what do you mean by `factorized with getVideoTrackInfo ?`

Reply:
Just move allocation of videotracks inside   if (isFormatSupported(format)) { 
To avoid unnecessary operation.  A simple point in fact :)

Reply:
Lol, didn't notice that, this was really dumb 🤦 😆 Fixed it 👍